ksqldb-engine/src/main/java/io/confluent/ksql/utilization/PersistentQuerySaturationMetrics.java

```diff
@@ -88,7 +88,8 @@ public PersistentQuerySaturationMetrics(
     this.reporter = Objects.requireNonNull(reporter, "reporter");
     this.window = Objects.requireNonNull(window, "window");
     this.sampleMargin = Objects.requireNonNull(sampleMargin, "sampleMargin");
-    this.customTags = Objects.requireNonNull(customTags, "customTags");
+    PersistentQuerySaturationMetrics.customTags =
```

Comment:
This is correct to fix the warning.  The fact that shared static state is updated gives me pause.

Is it worth thinking about this?  I'm unclear on how this class is used.

Reply:
Yeah -- my understanding is, that it's actually a singleton (so it "ok", meaning nothing breaks; not "ok" from a code layout POV) -- I did not want to do more refactoring to convert it into a real singleton. Might be best to refactor in a follow up PR?

\cc @rodesai @lct45 who worked on this code.

Reply:
Thanks @mjsax for the crash course on a singleton. @rodesai may be more equipped to answer this fully but yeah this is essentially a singleton without proper singleton implementation. We create this one in the `KsqlRestApplication` as a single reporter for these metrics and as far as I know, we wouldn't need a second instance

Reply:
I talked to Rohan and he proposed to rewrite the code not make the field non-static. I'll just update this PR accordingly.